### ISSUE TITLE
Hydration error on fresh install / LiveReload

### ISSUE THREAD

**Issue by wenzf** [remix-run/remix#6418]
### What version of Remix are you using?

1.16.1

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

1. npx create-remix@latest
choices:
- Just the basics
- Architect (AWS Lambda)
- Typescript
-  intsall via NPM

2. npm run dev 

Note: same behaviour after build or viewed in private mode. 

### Expected Behavior

start witouth errors

### Actual Behavior

failed hydration

`Warning: Expected server HTML to contain a matching <script> in <body>`

`Uncaught Error: Hydration failed because the initial UI does not match what was rendered on the server.`

No errors when `<LiveReload />` is removed from root.tsx

![Screenshot from 2023-05-18 10-39-12](https://github.com/remix-run/remix/assets/39794408/9d06184f-ebeb-448f-81ae-36006803544f)

**Comment by nemonemi** [comment]
I can confirm this behavior as well. Fresh install and this error.

**Comment by quangv** [comment]
Could be related to: https://github.com/remix-run/remix/issues/4822#issuecomment-1345851374

**Comment by dyankov91** [comment]
I'm facing the same issue. It doesn't seem to be related to what @quangv points to. I've tried using a browser without any extensions (Safari, Chrome, Firefox) and facing the same problem. I've tried using stacks or installing just the basis.

**Comment by wenzf** [comment]
> I'm facing the same issue. It doesn't seem to be related to what @quangv points to. I've tried using a browser without any extensions (Safari, Chrome, Firefox) and facing the same problem. I've tried using stacks or installing just the basis.

I also don't think it's related to that. I tested in several browsers (OS: Linux/Ubuntu) and in private mode. In another project I'm using v 1.14.1, there are no issues.

**Comment by dyankov91** [comment]
Okay, now I have a better understanding of the problem after some experimenting. 

The problem is caused inside `remix.config.js:4` by this line: `publicPath: "/_static/build/",`. It seems that something is not configured correctly and it causes the server and browser build to mismatch. 

Commenting out this line (or using the default value `/build/`) removes the hydration errors, but causes (or uncovers) another one. 

<img width="660" alt="not-found 2" src="https://github.com/remix-run/remix/assets/1302460/26f86408-d56c-424d-a272-74ba4ed505fc">


After commenting out this line I'm getting 404 on all assets in the `public/build` folder. I've tried playing around with `assetsBuildDirectory`, changing it to `/public/_static/build/` or `/_static/build/` trying to match the new value of the `publicPath` and while the files are generated on the new location they are still inaccessible on FE (404). 

I've tried placing a `.png` file inside the public folder and then tried to access it directly in the browser but I'm getting 404 and this logged out on the console `No routes matched location "/image.png"`. 

I can reproduce the exact same problem (404 on the assets) using the new `v2_routeConvention` flag or without it. 

Basically, now I'm getting the server build correctly but I cannot load the assets. Any idea, how to solve this?

**Comment by wenzf** [comment]
> Basically, now I'm getting the server build correctly but I cannot load the assets. Any idea, how to solve this?

I removed <LiveReload /> from the root. It's not a solution but works for me until it's fixed. - And if not, it's not a big drama since it seems purely related to development.

**Comment by jupiter229** [comment]
Same problem happens to me with a fresh indie-stack. Im using react 18, and remix v2, any updates here?

**Comment by kiliman** [comment]
See https://github.com/kiliman/remix-hydration-fix for an explanation of what is causing the React hydration errors. You can try one of the workarounds or update React to the latest canary version. They have fixed the hydration problem and will be in React v18.3.

`npm i react@canary react-dom@canary`

**Comment by kiliman** [comment]
Yes. I haven't had any hydration issues with the next version of React.

**Comment by pcattori** [comment]
Architect template has been improved to add support for `v2_dev` and in the process this bug was fixed. No longer reproducible on 1.19.2
